dataland-api-key-manager/src/main/kotlin/org/dataland/datalandapikeymanager/api/ApiKeyAPI.kt

```diff
@@ -0,0 +1,91 @@
+package org.dataland.datalandapikeymanager.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.dataland.datalandapikeymanager.model.ApiKeyAndMetaInfo
+import org.dataland.datalandapikeymanager.model.ApiKeyMetaInfo
+import org.dataland.datalandapikeymanager.model.RevokeApiKeyResponse
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestParam
+import javax.validation.constraints.Positive
+
+/**
+ * Defines the restful api-key-manager API.
+ */
+interface ApiKeyAPI {
+
+    @Operation(
+        summary = "Request a new API key with expiration date and associated.",
+        description = "Requests a new API key with expiration date associated for the logged in user."
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved new api key.")
+        ]
+    )
+    @GetMapping(
+        value = ["/generateApiKey"],
+        produces = ["application/json"]
+    )
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @SecurityRequirement(name = "default-bearer-auth")
+    @SecurityRequirement(name = "default-oauth")
+    /**
+     * A method to generate a new API key
+     * @param daysValid int determining how many days the generated API key can be used
+     * @return new API key for the user
+     */
+    fun generateApiKey(
+        @RequestParam(required = false) @Positive daysValid: Int? = null
```

Comment:
Finde ich gut das hier direkt zu erzwingen dann kann die Logik im controller auch weg

Reply:
Achso durch das positive kann gar keine negative Zahl eingetragen werden.
Das sehe ich jetzt erst.
Ich prüfe jetzt mal, ob ich den "keine-Negativzahl" Check im Code rausnehmen kann und alles weiterhin wie erwartet funktioniert.

Reply:
Ok bei meinem ersten Versuch jetzt hat die @Positive annotation nicht verhindert, dass ich eine negative Zahl mitgebe.

Ich recherchier nochmal ein bisschen dazu.

Reply:
Wäre natürlich elegant gewesen aber wenn das nicht geht finde ich das so wie es jetzt ist auch in Ordnung. Das wichtige war für mich das die Logik aus dem controller verschwindet.